### DISCUSSION TITLE
Cannot import name 'packaging' from 'pkg_resources'

### DISCUSSION THREAD

**Discussion by MatteoAbrate** [Xilinx/finn#1121] [Q&A] [answered]
Hi everyone, I'm once again struggling with FINN. 
I managed to install ubuntu 18.04 on a virtualbox istance and inside it i'm creating the docker env making the pull inside a folder in the desktop. As soon as I'm trying to do anything with FINN this error pops up:

ImportError: cannot import name 'packaging' from 'pkg_resources' (/usr/local/lib/python3.10/dist-packages/pkg_resources/__init__.py)

it is pointing to the import section of the scripts, searching online the only thing i found was that this error can be caused by setuptools=70, so i downgrade it to the 69.5.1 version, than 69.0 than 68.0 but none of them worked. What could i do? the last time i run the exact same jupyter notebook it runs without problems so i'm thinking that it is something related to the installation or some dependancy.

Please help me.
Thanks in advance

Matteo Abrate (Turin, Italy, Department of Physics)

**Comment by fpjentzsch** [comment, marked as answer]
Hi,

I believe this was fixed in https://github.com/Xilinx/finn/pull/1102. Can you try the latest dev branch?

**Reply by iyinolu** [reply]
This fixed the quicktest failures but I had a problem with accessing the jupyter notebook server on the dev branch. The generated token was always invalid.

Running `jupyter server list` inside the container returned as empty.

I'm running this with ubuntu via WSL

**Reply by fpjentzsch** [reply]
Did you make sure that no other containers/Jupyter servers are running in the background? For me it works if I run `./run-docker.sh notebook` and click on the link it gives me: "http://127.0.0.1:8888/tree?token=..."

**Reply by iyinolu** [reply]
Yes, that's correct. It works! I had a jupyter lab running in the background. Thanks
